src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -106,13 +114,62 @@ def check_lazy(*args: Any, **kwargs: Any) -> Any:
 
 
 class Index(metaclass=TelemetryMeta):
+
+    # Equivalent index type in native pandas
+    _NATIVE_INDEX_TYPE = native_pd.Index
+
+    def __new__(
+        cls,
+        data: ArrayLike | SnowflakeQueryCompiler | None = None,
+        dtype: str | np.dtype | ExtensionDtype | None = _CONSTRUCTOR_DEFAULTS["dtype"],
+        copy: bool = _CONSTRUCTOR_DEFAULTS["copy"],
+        name: object = _CONSTRUCTOR_DEFAULTS["name"],
+        tupleize_cols: bool = _CONSTRUCTOR_DEFAULTS["tupleize_cols"],
+        convert_to_lazy: bool = True,
+    ) -> Index:
+        """
+        Override __new__ method to control new instance creation of Index.
+        Depending of data type, it will create a Index or DatetimeIndex instance.
+
+        Parameters
+        ----------
+        data : array-like (1-dimensional)
+        dtype : str, numpy.dtype, or ExtensionDtype, optional
+            Data type for the output Index. If not specified, this will be
+            inferred from `data`.
+            See the :ref:`user guide <basics.dtypes>` for more usages.
+        copy : bool, default False
+            Copy input data.
+        name : object
+            Name to be stored in the index.
+        tupleize_cols : bool (default: True)
+            When True, attempt to create a MultiIndex if possible.
+        convert_to_lazy : bool (default: True)
+            When True, create a lazy index object from a local data input, otherwise, create an index object that saves a pandas index locally.
+            We only set convert_to_lazy as False to avoid pulling data back and forth from Snowflake, e.g., when calling df.columns, the column data should always be kept locally.
+        """
+        from snowflake.snowpark.modin.plugin.extensions.datetime_index import (
+            DatetimeIndex,
+        )
+
+        if isinstance(data, SnowflakeQueryCompiler):
```

Comment:
can you add a comment here that a DatetimeIndex object will be returned if the column/date have datetime type?

Reply:
Ack